### DISCUSSION TITLE
Generate JSON file with content based on C# source code during build

### DISCUSSION THREAD

**Discussion by jozefizso** [dotnet/roslyn#76323] [Q&A] [answered]
Hi, we are building a custom tool to register COM classes during development. We would like to generate a JSON file with list of COM classes defined in the project source code so we can use this information later to register COM classes.

Is it possile to generate a temporary files to `obj/` output using Roslyn's source code generation?

PS: we cannot use `regasm.exe` as it requires admin rights to register classes. We are registering classes to user's registry and also writing required key for Microsoft Office apps to register add-in classes. We also need a solution to work with both .NET Framework and .NET Core assemblies.

See also https://github.com/dotnet/msbuild/discussions/8650

**Comment by 333fred** [comment, marked as answer]
No. Source generators can only generate C# source. See https://github.com/dotnet/roslyn/issues/49935.

**Comment by Misiu** [comment]
The only option for now is a custom MSTask, but I fully agree that source generators should support not only C# code output.
Ref: https://github.com/dotnet/roslyn/issues/57608
